docs/smart-contracts/guides/transactions/simulateTransaction-Deep-Dive.mdx

```diff
@@ -0,0 +1,403 @@
+
+- **Read-Only Keys**: Available for reading only.
+- **Read-Write Keys**: Available for reading and writing.
+
+The footprint ensures that a transaction is aware of all the ledger entries it will interact with, preventing unexpected errors during execution.
+
+## Assembling a Transaction
+
+Once you have simulated the transaction and obtained the necessary data, you can assemble the transaction for actual submission. The `assembleTransaction` function in the SDK helps with this process, but you can also call `prepareTransaction` to have it both simulate and assemble the transaction for you.
+
+### js-stellar-sdk `assembleTransaction `
+
+Using the javascript SDK we can call [`assembleTransaction`](https://github.com/stellar/js-stellar-sdk/blob/master/src/rpc/transaction.ts#L28) to easily assemble a transaction.
+
+## Handling Archived Ledger Entries
+
+When a ledger entry is archived, it needs to be restored before the transaction can be submitted. This is indicated in the `restorePreamble` field of the result.
+
+```typescript
+interface RestorePreamble {
+  minResourceFee: string; // Recommended minimum resource fee to add when submitting the RestoreFootprint operation.
+  transactionData: string; // The recommended Soroban Transaction Data to use when submitting the RestoreFootprint operation.
+}
+```
+
+Here is an example for handling restoration using the `restorePreamble` to restore archived data:
+
+```typescript
+/**
+ * Simulates a restoration transaction to determine if restoration is needed.
+ * This function first checks the ledger entry for the given WASM hash. If the entry is found and has expired,
+ * it attempts a restoration. If the entry hasn't expired yet but the TTL needs extension, it proceeds with TTL extension.
+ * @param contract - The address of the contract to check
+ * @param txParams - The transaction parameters including account and signer.
+ * @returns A promise that resolves to a simulation response, indicating whether restoration or TTL extension is needed.
+ */
+export async function simulateRestorationIfNeeded(
+  contract: ContractAddress,
+  txParams: TxParams,
+): Promise<
+  SorobanRpc.Api.SimulateTransactionRestoreResponse | string | undefined
+> {
+  try {
+    const account = await config.rpc.getAccount(txParams.account.accountId());
+    const contract = new Contract(contract);
+    //const ledgerKey = getLedgerKeyWasmId(wasmHash);
+    const ledgerKey = contract.getFootprint();
+    const response = await config.rpc.getLedgerEntries(...[ledgerKey]);
+    if (
+      response.entries &&
+      response.entries.length > 0 &&
+      response.entries[0].liveUntilLedgerSeq
+    ) {
+      const expirationLedger = response.entries[0].liveUntilLedgerSeq;
+      const desiredLedgerSeq = response.latestLedger + 500000;
+      let extendLedgers = desiredLedgerSeq - expirationLedger;
+      if (extendLedgers < 10000) {
+        extendLedgers = 10000;
+      }
```

Comment:
definitely worth highlighting the tradeoff here in comments: 500k ledgers of expiry is $$$